File: Dsa-day116.py
class Solution:
    def minOperations(self, nums: List[int]) -> int:
        output = 0
        stack = []
        for num in nums:
            while stack and stack[-1] > num:
                output += 1
                stack.pop()
            if num and (not stack or stack[-1] != num):
                stack.append(num)
        return output + len(stack)

File: Dsa-day116.java
class Solution {
    public int minOperations(int[] nums) {
        int output = 0;
        Deque<Integer> stack = new ArrayDeque<>();
        for (int num : nums) {
            while (!stack.isEmpty() && stack.peekLast() > num) {
                output++;
                stack.pollLast();
            }
            if (num != 0 && (stack.isEmpty() || stack.peekLast() != num))
                stack.addLast(num);
        }
        return output + stack.size();
    }
}
